src/System.Text.Json/src/System/Text/Json/Document/JsonDocument.Parse.cs

```diff
@@ -554,17 +554,12 @@ private static bool TryParseValue(ref Utf8JsonReader reader, out JsonDocument do
             byte[] extraRentedBytes)
         {
             ReadOnlySpan<byte> utf8JsonSpan = utf8Json.Span;
-            Utf8JsonReader reader = new Utf8JsonReader(
-                utf8JsonSpan,
-                isFinalBlock: true,
-                new JsonReaderState(options: readerOptions));
-
             var database = new MetadataDb(utf8Json.Length);
             var stack = new StackRowStack(JsonDocumentOptions.DefaultMaxDepth * StackRow.Size);
 
             try
             {
-                Parse(utf8JsonSpan, reader, ref database, ref stack);
```

Comment:
I think we should port this to 3.0, particularly passing a mutable struct by value here (the `Utf8JsonReader`). Aside from reducing the struct copy/perf benefit, it would be best if we reduce accidental pass-by-value code (to avoid false positives when looking for such instances in the future).